lib/src/model/write_characteristic_info.dart

```diff
@@ -1,17 +1,33 @@
 import 'package:flutter_reactive_ble/src/model/generic_failure.dart';
 import 'package:flutter_reactive_ble/src/model/qualified_characteristic.dart';
 import 'package:flutter_reactive_ble/src/model/result.dart';
+import 'package:flutter_reactive_ble/src/model/unit.dart';
 import 'package:meta/meta.dart';
 
 @immutable
 class WriteCharacteristicInfo {
   final QualifiedCharacteristic characteristic;
-  final Result<void, GenericFailure<WriteCharacteristicFailure>> result;
+  final Result<Unit, GenericFailure<WriteCharacteristicFailure>?> result;
```

Comment:
You see? When you need it, you can explicitly _specialize_ `Result` with optional types. `Result` itself should not add optionality under the hood.

Reply:
What do you exactly mean by this? the optionality is linked to the failure

Reply:
I made a few comments related to `result.dart` and this one implicitly refers to them.

In particular, you have made both the `_value` and `_failure` fields in `Result` _optional_, which should not be done, because when you really want one or both of them optional, you can specialize the class with optional types, _as you are doing here_.